modules/twinklexfd.js

```diff
@@ -346,53 +346,43 @@ Twinkle.xfd.callback.change_category = function twinklexfdCallbackChangeCategory
-				'Geographic/Europe/United Kingdom': ['United Kingdom', 'England', 'Northern Ireland', 'Scotland', 'Wales'],
-				'Geographic/Oceania': ['Oceania', 'Antarctica', 'Australia', 'New Zealand'],
-				'Geographic/Americas/Canada': ['Canada', 'Alberta', 'British Columbia', 'Manitoba', 'Nova Scotia', 'Ontario', 'Quebec'],
-				'Geographic/Americas/Latin America': ['Latin America', 'Caribbean', 'South America', 'Argentina', 'Barbados', 'Belize', 'Bolivia', 'Brazil', 'Chile', 'Colombia', 'Costa Rica', 'Cuba', 'Ecuador', 'El Salvador', 'Grenada', 'Guatemala', 'Haiti', 'Mexico', 'Nicaragua', 'Panama', 'Paraguay', 'Peru', 'Puerto Rico', 'Trinidad and Tobago', 'Uruguay', 'Venezuela'],
-				'Geographic/Americas/USA': ['United States of America', 'Alabama', 'Alaska', 'Arizona', 'Arkansas', 'California', 'Colorado', 'Connecticut', 'Delaware', 'Florida', 'Georgia (U.S. state)', 'Hawaii', 'Idaho', 'Illinois', 'Indiana', 'Iowa', 'Kansas', 'Kentucky', 'Louisiana', 'Maine', 'Maryland', 'Massachusetts', 'Michigan', 'Minnesota', 'Mississippi', 'Missouri', 'Montana', 'Nebraska', 'Nevada', 'New Hampshire', 'New Jersey', 'New Mexico', 'New York', 'North Carolina', 'North Dakota', 'Ohio', 'Oklahoma', 'Oregon', 'Pennsylvania', 'Rhode Island', 'South Carolina', 'South Dakota', 'Tennessee', 'Texas', 'Utah', 'Vermont', 'Virginia', 'Washington', 'Washington, D.C.', 'West Virginia', 'Wisconsin', 'Wyoming'],
-				'Geographic/Unsorted': ['Islands']
-			};
-
-			var delsort = work_area.append({
+			work_area.append({
 				type: 'select',
 				multiple: true,
 				name: 'delsortCats',
 				label: 'Choose deletion sorting categories:',
 				tooltip: 'Select a few categories that are specifically relevant to the subject of the article. Be as precise as possible; categories like People and USA should only be used when no other categories apply.'
 			});
 
-			$.each(delsortCategories, function(groupname, list) {
-				var group = delsort.append({ type: 'optgroup', label: groupname });
-				list.forEach(function(item) {
-					group.append({ type: 'option', label: item, value: item });
-				});
-			});
+			// delsort categories taken from [[Wikipedia:WikiProject Deletion sorting/Computer-readable.json]]
+			var query = {
+				action: 'parse',
+				format: 'json',
+				page: 'Wikipedia:WikiProject_Deletion_sorting/Computer-readable.json',
+				prop: 'wikitext',
+				smaxage: '86400', // cache for 1 day
+				maxage: '86400' // cache for 1 day
+			};
+			new Morebits.wiki.api('Get JSON list of deletion sorting categories', query, function(apiobj) {
+				try {
+					var wikitext = apiobj.getResponse().parse.wikitext;
+					var delsortCategories = JSON.parse(wikitext);
+					var select = $('[name="delsortCats"]');
```

Comment:
```suggestion
					var $select = $('[name="delsortCats"]');
```
Per convention, jQuery vars have a $ prefix.